nativelib/src/main/resources/pwd.c

```diff
@@ -20,17 +21,19 @@ void scalanative_passwd_copy(struct passwd *passwd,
 }
 int scalanative_getpwuid(scalanative_uid_t uid,
                          struct scalanative_passwd *buf) {
+    errno = 0;
     struct passwd *passwd = getpwuid(uid);
 
     if (passwd == NULL) {
-        return 1;
+        return errno == 0 ? 1 : -1;
     } else {
         scalanative_passwd_copy(passwd, buf);
         return 0;
     }
 }
 
 int scalanative_getpwnam(char *name, struct scalanative_passwd *buf) {
+    errno = 0;
```

Comment:
Unless I am missing something, all three edits in the file should be deleted.
The clearing of errno could be author's perogative

Changing the  non-standard handling in the getpwuid code from two values to three valued and
then never checking anything other than the 0 result seems to add complexity with little benefit.

Reply:
Yes, you're right. I just forget to revert this file too besides `grp.c`.

Reply:
Fixed.